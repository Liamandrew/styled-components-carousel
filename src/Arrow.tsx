import React from 'react';
import styled from 'styled-components';

export enum ArrowType {
    previous = 'previous',
    next = 'next',
}

type Props = {
    type: ArrowType;
    disabled?: boolean;
    onClick: () => void;
};

const Arrow = styled.button<Partial<Props>>`
    width: 32px;
    height: 32px;
    padding: 8px;
    border: 1px solid #DDDDDD;
    border-radius: 50%;
    background-color: ${props => (props.disabled ? '#F3F3F3' : '#FFFFFF')};
`;

const Component: React.FC<Props> = ({ type, onClick, disabled }) =>
    type === ArrowType.previous ? (
        <Arrow disabled={disabled} onClick={onClick}>
            &#8249;
        </Arrow>
    ) : (
        <Arrow disabled={disabled} onClick={onClick}>
            &#8250;
        </Arrow>
    );

export default Component;
