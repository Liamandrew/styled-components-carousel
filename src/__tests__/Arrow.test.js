import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Arrow from '../Arrow';

describe('Arrow', () => {
    const onClick = jest.fn();

    beforeEach(() => {
        jest.clearAllMocks();
    });

    it('should match the snapshot for Previous arrow', () => {
        const { container } = render(<Arrow type="previous" />);

        expect(container).toMatchSnapshot();
    });

    it('should match the snapshot for Next arrow', () => {
        const { container } = render(<Arrow type="next" />);

        expect(container).toMatchSnapshot();
    });

    it('should call onClick when enabled and clicked', () => {
        const { container } = render(<Arrow type="next" onClick={onClick} />);

        fireEvent.click(container.querySelector('button'));

        expect(onClick).toHaveBeenCalled();
    });

    it('should not call onClick when disabled and clicked', () => {
        const { container } = render(<Arrow type="next" onClick={onClick} disabled />);

        fireEvent.click(container.querySelector('button'));

        expect(onClick).not.toHaveBeenCalled();
    });
});
