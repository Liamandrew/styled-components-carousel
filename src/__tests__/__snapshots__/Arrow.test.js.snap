// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arrow should match the snapshot for Next arrow 1`] = `
<div>
  <button
    class="sc-AxjAm dPRmsH"
    data-testid="arrow-next"
  >
    ›
  </button>
</div>
`;

exports[`Arrow should match the snapshot for Previous arrow 1`] = `
<div>
  <button
    class="sc-AxjAm dPRmsH"
    data-testid="arrow-previous"
  >
    ‹
  </button>
</div>
`;
